pkg/crc/services/dns/dns_darwin.go

```diff
@@ -37,22 +37,41 @@ func runPostStartForOS(serviceConfig services.ServicePostStartConfig, result *se
 	}
 
 	// Write resolver config to host
-	if err := createResolverFile(serviceConfig.IP, filepath.Join("/", "etc", "resolver", serviceConfig.BundleMetadata.ClusterInfo.BaseDomain)); err != nil {
+	needRestart, err := createResolverFile(serviceConfig.IP, filepath.Join("/", "etc", "resolver", serviceConfig.BundleMetadata.ClusterInfo.BaseDomain))
+	if err != nil {
 		result.Success = false
 		return *result, err
 	}
-	// Restart the Network on mac
-	logging.InfoF("Restarting the host network")
-	if err := restartNetwork(); err != nil {
-		result.Success = false
-		return *result, err
+	if needRestart {
+		// Restart the Network on mac
+		logging.InfoF("Restarting the host network")
+		if err := restartNetwork(); err != nil {
+			result.Success = false
+			return *result, err
+		}
+	} else {
+		logging.InfoF("Network restart not needed")
 	}
+
 	// we pass the result and error on
 	result.Success = true
 	return *result, nil
 }
 
-func createResolverFile(InstanceIP string, path string) error {
+func writeIfContentChanged(path string, new_content []byte) (bool, error) {
```

Comment:
Can we extract this function and have it part of `pkg/os/util.go` so we can reuse it in future if we required that kind of functionality and also name as `WriteIfFileContentChanged`

Reply:
Done, I named it `WriteFileIfContentChanged` and added a `perm` argument to be similar to ioutil.WriteFile.